src/ZfrCors/Factory/CorsServiceFactory.php

```diff
@@ -35,11 +36,20 @@ class CorsServiceFactory implements FactoryInterface
      * {@inheritDoc}
      * @return CorsService
      */
-    public function createService(ServiceLocatorInterface $serviceLocator)
+    public function __invoke(ContainerInterface $container, $requestedName, array $options = null)
     {
         /* @var $corsOptions CorsOptions */
-        $corsOptions = $serviceLocator->get('ZfrCors\Options\CorsOptions');
+        $corsOptions = $container->get('ZfrCors\Options\CorsOptions');
 
         return new CorsService($corsOptions);
     }
+    /**
+     * {@inheritDoc}
+     *
+     * @return CorsService
+     */
+    public function createService(ServiceLocatorInterface $container)
```

Comment:
@weierophinney  I don't know apigility and  didn't used it but..
> What @macnibblet is suggesting is the route we went with factories in Apigility itself: they do not implement FactoryInterface, but instead are simply callable; additionally, we don't typehint the container. This provides forwards compatibility out-of-the-box.

not related, but you forgot about plugin managers and  I guess factories implementing MutableCreationOptionsInterface they also will not work out-of-the-box.

I did those changes the way documented in migration documentation but I guess some modules does things differently. :D